src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -310,6 +344,23 @@ def sync_deploy_root_with_stage(self, role: str) -> DiffResult:
             % self.deploy_root
         )
         diff: DiffResult = stage_diff(self.deploy_root, self.stage_fqn)
+
+        # If we are syncing specific files/directories, remove everything else from the diff
+        if len(paths_to_sync) > 0:
+            paths_to_sync = [resolve_without_follow(p) for p in paths_to_sync]
+            if not recursive:
+                _verify_no_directories(paths_to_sync)
+            deploy_paths = [
+                project_path_to_deploy_path(p, created_files) for p in paths_to_sync
+            ]
+            paths_to_keep = set(
+                _get_files_to_sync(deploy_paths, self.deploy_root.resolve())
+            )
```

Comment:
ok, subjective, but the naming confuses me here... We have deploy_paths, paths_to_sync, paths_to_keep and files_to_sync. Can you replace the names with more descriptive options? I'm losing track of what's what.

Reply:
Changed everything to `paths_to_sync`